src/engine/openspaceengine.cpp

```diff
@@ -1546,6 +1584,144 @@ void OpenSpaceEngine::toggleShutdownMode() {
     }
 }
 
+void OpenSpaceEngine::setFromProfile_camera(const Profile& p) {
+    std::visit(
+        overloaded{
+            [](const Profile::CameraNavState& navStateProfile) {
+                interaction::NavigationState nav;
+                nav.anchor = navStateProfile.anchor;
+                if (navStateProfile.aim.has_value()) {
+                    nav.aim = navStateProfile.aim.value();
+                }
+                if (nav.referenceFrame.empty()) {
+                    nav.referenceFrame = "Root";
+                }
+                nav.position = navStateProfile.position;
+                if (navStateProfile.up.has_value()) {
+                    nav.up = navStateProfile.up;
+                }
+                if (navStateProfile.yaw.has_value()) {
+                    nav.yaw = navStateProfile.yaw.value();
+                }
+                if (navStateProfile.pitch.has_value()) {
+                    nav.pitch = navStateProfile.pitch.value();
+                }
+                global::navigationHandler->setNavigationStateNextFrame(nav);
+            },
+            [](const Profile::CameraGoToGeo& geo) {
+                std::string geoScript = fmt::format("openspace.globebrowsing.goToGeo"
+                    "([[{}]], {}, {}", geo.anchor, geo.latitude, geo.longitude);
+                if (geo.altitude.has_value()) {
+                    geoScript += fmt::format(", {}", geo.altitude.value());
+                }
+                geoScript += ")";
+                global::scriptEngine->queueScript(
+                    geoScript,
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        },
+        p.camera.value()
+    );
+}
+
+void OpenSpaceEngine::setFromProfile_deltaTimes(const Profile& p) {
+    global::timeManager->setDeltaTimeSteps(p.deltaTimes);
+}
+
+void OpenSpaceEngine::setFromProfile_modules(const Profile& p)
+{
+    for (Profile::Module mod : p.modules) {
+        const std::vector<OpenSpaceModule*>& m = global::moduleEngine->modules();
+        const auto it = std::find_if(m.begin(), m.end(),
+            [&mod](const OpenSpaceModule* moduleSearch) {
+                return (moduleSearch->identifier().compare(mod.name) == 0);
+            });
+        if (it != m.end()) {
+            if (mod.loadedInstruction.has_value()) {
+                global::scriptEngine->queueScript(
+                    mod.loadedInstruction.value(),
```

Comment:
To access the optional, you can just dereference it:
-> `*mod.loadedInstruction`

Same for the notLoadedInstruction below

Reply:
I like it this way